### PR TITLE
added resources to learn and practice concepts

### PR DESCRIPTION
**Pull request by ahoiberg**

### CODE REVIEW

**Comment by tekknolagi** [comment]
Do you think it's worth having leetcode and hackerrank on the main page? If you had to fill up a computer screen with Tufts CS related links, would those make the cut?

I think having xinyminutes for languages used in the dept (C, C++, SML, JS, etc) would be valuable.

**Comment by ahoiberg** [comment]
it’s up to you! I think the problems they ask are reasonably interesting, though their interfaces are crap. I like the idea of making specific learn xinyminutes for classes, I’ll add that now. I want to keep the home page there to encourage people to explore other languages though.

File: index.md [hunk 1]
```diff
@@ -41,3 +41,19 @@
...
+## Learn new things
+
+* [Learn X in Y minutes](https://learnxinyminutes.com/)
+* [Learn C in Y minutes (used in 40)](https://learnxinyminutes.com/docs/c/)
+* [Learn C++ in Y minutes (used in 11/15)](https://learnxinyminutes.com/docs/c++/)
+* [Learn Standard ML in Y minutes (used in 105)](https://learnxinyminutes.com/docs/standard-ml/)
+* [Learn Bash in Y minutes (helpful for navigating the terminal)](https://learnxinyminutes.com/docs/bash/)
+* [Learn Git in Y minutes (helpful for version control)](https://learnxinyminutes.com/docs/git/)
+* [Learn Vim in Y minutes (Vim is a wonderful text editor)](https://learnxinyminutes.com/docs/vim/)
+* [Learn LaTeX in Y minutes (great for formatting proofs)](https://learnxinyminutes.com/docs/latex/)
```
Comment (tekknolagi) [inline index.md]: Can you note that this is used in 160, 170, 105, etc

File: index.md [hunk 2]
```diff
@@ -41,3 +41,19 @@
 
 * [Interview cheat sheet](https://gist.github.com/TSiege/cbb0507082bb18ff7e4b)
 * [Google interview university](https://github.com/jwasham/google-interview-university)
+
+## Learn new things
+
+* [Learn X in Y minutes](https://learnxinyminutes.com/)
+* [Learn C in Y minutes (used in 40)](https://learnxinyminutes.com/docs/c/)
+* [Learn C++ in Y minutes (used in 11/15)](https://learnxinyminutes.com/docs/c++/)
+* [Learn Standard ML in Y minutes (used in 105)](https://learnxinyminutes.com/docs/standard-ml/)
```
Comment (tekknolagi) [inline index.md]: Can you re-order these so that they (roughly) reflect course orderings?

File: index.md [hunk 3]
```diff
@@ -41,3 +41,19 @@
 
 * [Interview cheat sheet](https://gist.github.com/TSiege/cbb0507082bb18ff7e4b)
 * [Google interview university](https://github.com/jwasham/google-interview-university)
+
+## Learn new things
+
+* [Learn X in Y minutes](https://learnxinyminutes.com/)
```
Comment (tekknolagi) [inline index.md:47]: Perhaps make the others sub-headings of this one

File: index.md [hunk 4]
```diff
@@ -41,3 +41,19 @@
...
+* [Learn C in Y minutes (used in 40)](https://learnxinyminutes.com/docs/c/)
+* [Learn C++ in Y minutes (used in 11/15)](https://learnxinyminutes.com/docs/c++/)
+* [Learn Standard ML in Y minutes (used in 105)](https://learnxinyminutes.com/docs/standard-ml/)
+* [Learn Bash in Y minutes (helpful for navigating the terminal)](https://learnxinyminutes.com/docs/bash/)
+* [Learn Git in Y minutes (helpful for version control)](https://learnxinyminutes.com/docs/git/)
+* [Learn Vim in Y minutes (Vim is a wonderful text editor)](https://learnxinyminutes.com/docs/vim/)
```
Comment (tekknolagi) [inline index.md]: Is there Emacs as well? I think COMP 15 teaches Emacs at the moment.

File: index.md [hunk 5]
```diff
@@ -41,3 +41,19 @@
...
+* [Learn LaTeX in Y minutes (great for formatting proofs)](https://learnxinyminutes.com/docs/latex/)
+
+## Practice, Practice, Practice
```
Comment (tekknolagi) [inline index.md]: Why? What's this good for? Will the majority of the dept find this useful?

File: index.md (hunk 3 above)
Comment (tekknolagi) [reply index.md:47]: @ahoiberg https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#lists

**Pull request merged**